lib/hcloud/volume.rb

```diff
@@ -26,6 +28,8 @@ def detach
     end
 
     def resize(size:)
+      raise Hcloud::Error::InvalidInput, 'no size given' if size.nil?
```

Comment:
size has to be an int (and actually greater that the current size). So I would change that check a bit to something like
```
   raise Hcloud::Error::InvalidInput, 'invalid size given' unless new_size.to_i > size
```